core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala

```diff
@@ -206,8 +228,7 @@ private[spark] class CoarseMesosSchedulerBackend(
             Collections.singleton(offer.getId), Collections.singletonList(task.build()), filters)
         } else {
           // Filter it out
-          d.launchTasks(
-            Collections.singleton(offer.getId), Collections.emptyList[MesosTaskInfo](), filters)
+          d.declineOffer(offer.getId)
```

Comment:
It's the same thing, just clearer in its intentions. According to Javadocs on [SchedulerDriver](http://mesos.apache.org/api/latest/java/org/apache/mesos/SchedulerDriver.html#launchTasks(java.util.Collection,%20java.util.Collection,%20Filters)):

> .. Invoking this function with an empty collection of tasks declines offers in their entirety (see declineOffer(OfferID, Filters)).